usvm-core/src/main/kotlin/org/usvm/constraints/PathConstraints.kt

```diff
@@ -150,23 +164,28 @@ open class UPathConstraints<Type>(
 
                         notConstraint is UOrExpr -> notConstraint.args.forEach { plusAssign(ctx.mkNot(it)) }
 
-                        else -> logicalConstraints += constraint
+                        else -> logicalConstraints.add(constraint, ownership)
                     }
                 }
 
                 logicalConstraints.contains(constraint.not()) -> contradiction(ctx)
 
-                else -> logicalConstraints += constraint
+                else -> logicalConstraints.add(constraint, ownership)
             }
         }
 
-    open fun clone(): UPathConstraints<Type> {
+    open fun clone(
+        thisOwnership: MutabilityOwnership = ownership,
```

Comment:
I don't like default `ownership` value for `thisOwnership`. This can lead to incorrect states. I prefer to introduce something like scoped ownership: we change both ownerships, this and cloned, we perform operation on cloned, if we don't need cloned anymore we rollback this ownership.

Reply:
It seems like checked constraints with temporary fresh ownership are satisfiable, we don't have to manually change  both ownerships in `state.clone`. So we can preserve temporary ownership, if we are not going to use ownership from constraints